projects/sample-widget/src/lib/main.ts

```diff
@@ -49,52 +118,212 @@ template.innerHTML = `
+
+    api.searchValue$.subscribe((value) => {
       box.innerHTML = value;
-    })
+    });
+  }
+
+  createProcessEl(
+    process: IProcess,
+    jobStatus: number,
+    jobIdentifier?: string
+  ) {
+    // TODO: replace with enum
+    const processLetter = document.createElement("div") as HTMLElement;
+    processLetter.innerHTML = process.initialLetter;
+    processLetter.style.color = `var(--icons-tile-letter-color-${process.color})`;
+    processLetter.className = "process-letter";
+
+    const processName = document.createElement("p");
+    processName.innerHTML = process.name;
+    processName.className = "process-name";
+
+    const processVersion = document.createElement("p");
+    processVersion.className = "process-version";
+    processVersion.innerHTML = `Version ${process.version}`;
+
+    const processNameAndVersion = document.createElement("div");
+    processNameAndVersion.className = "process-name-and-version";
+    processNameAndVersion.appendChild(processName);
+    processNameAndVersion.appendChild(processVersion);
+
+    const processButtons = document.createElement("div");
+    processButtons.className = "process-buttons";
+
+    if (process.state !== 0) {
+      // TODO: Change to ProcessInstallationState.Installed
+      const runButton = document.createElement("button");
+      runButton.className = `run-button run-button-${process.key}`;
+      runButton.innerHTML = this.runSvg;
+      runButton.style.display = this.getRunButtonDisplay(jobStatus);
+      runButton.addEventListener("click", () => this.startJob(process.key));
+      processButtons.appendChild(runButton);
+
+      const stopButton = document.createElement("button");
+      stopButton.className = `stop-button stop-button-${process.key}`;
+      stopButton.innerHTML = this.stopSvg;
+      stopButton.style.display = this.getStopButtonDisplay(jobStatus);
+      stopButton.addEventListener("click", () => this.stopJob(jobIdentifier));
+      processButtons.appendChild(stopButton);
+    } else {
+      const installButton = document.createElement("button");
+      installButton.className = `install-button install-button-${process.key}`;
+      installButton.innerHTML = this.installSvg;
+      installButton.addEventListener("click", () =>
+        this.installProcess(process.key)
+      );
+      processButtons.appendChild(installButton);
+    }
+
+    const processContainer = document.createElement("div");
+    processContainer.className = "process";
+    processContainer.appendChild(processLetter);
+    processContainer.appendChild(processNameAndVersion);
+    processContainer.appendChild(processButtons);
+
+    return processContainer;
+  }
+
+  startJob(processKey: string) {
+    (api as any) // TODO: remove any
+      .startJob({
+        args: {},
+        connections: {},
+        ORConnectionOverrides: { resourceOverwrites: {} },
+        processKey: processKey,
+        source: "ProcessListButton",
```

Comment:
I think it would be more useful for our telemetry to always override this field with something generic like 'ExternalWidget' on Assistant's side. That way we can have a clear view of how many times jobs are started from these widgets, and the widget devs don't have to worry about a field that's currently mandatory and makes no difference to them (so they're likely to input some dummy name).